BitwardenShared/UI/Auth/Extensions/AuthRouter+Redirects.swift

```diff
@@ -361,11 +370,14 @@ extension AuthRouter {
                     return .landingSoftLoggedOut(email: activeAccount.profile.email)
                 }
 
+                let rehydratableTarget = try await services.rehydrationHelper.getSavedRehydratableTarget()
+
                 return .vaultUnlock(
                     activeAccount,
                     animated: animated,
                     attemptAutomaticBiometricUnlock: attemptAutomaticBiometricUnlock,
-                    didSwitchAccountAutomatically: didSwitchAccountAutomatically
+                    didSwitchAccountAutomatically: didSwitchAccountAutomatically,
+                    rehydratableTarget: rehydratableTarget
```

Comment:
⚠️ I think there might be a bug here. The `rehydratableTarget` is being set as the auth completion route when you navigate to the vault unlock screen, before the vault is unlocked. If you then switch accounts and unlock that vault, the app will try to rehydrate that route, for the wrong user.

It might be better of to do this in `AuthRouter.completeAuthRedirect()`. That occurs after the user's vault is unlocked, right before navigating to the vault. We could also pass the route to restore as part of the `.complete` `AuthRoute`. The `AuthCoordinator` then calls `delegate?.didCompleteAuth()`, which I think we could use to pass that route and then we wouldn't have to add a separate `setAuthCompletionRoute` delegate method?

https://github.com/bitwarden/ios/blob/e21b142444e0e631de70cbd395593f8f647b09e7/BitwardenShared/UI/Auth/Extensions/AuthRouter%2BRedirects.swift#L27-L39

Reply:
Excellent point, thanks for raising this! And as further discussed, changed the approach so the logic is done in `completeAuthRedirect()` and added a new route `.completeWithRehydration(rehydratableTarget)` which passes the target using `delegate?.didCompleteAuth(rehydratableTarget)` to perform the correct navigation after auth completes.